src/main/java/com/cognifide/aemrules/checks/ContentResourceShouldBeNullCheckedCheck.java

```diff
@@ -0,0 +1,261 @@
+            if (method.expression() instanceof MethodInvocationTree) {
+                MethodInvocationTree invocation = (MethodInvocationTree) method.expression();
+                if (!invocation.symbol().isUnknown() &&
+                    isPage(invocation.symbol().owner().type().fullyQualifiedName()) &&
+                    isGetContentResourceUsedOnPage(invocation) &&
+                    !returnOccurredInsideEqualNullCheck) {
+                    context.reportIssue(this, tree, RULE_MESSAGE);
+                }
+            }
+        }
+    }
+
+    private boolean isResource(AssignmentExpressionTree tree) {
+        return Constants.RESOURCE_TYPE.equals(tree.variable().symbolType().fullyQualifiedName());
+    }
+
+    private boolean isResourceInitialized(VariableTree tree) {
+        return tree.initializer() != null && tree.initializer().kind() != Kind.NULL_LITERAL;
+    }
+
+    private boolean isGetContentResourceUsedOnPage(MethodInvocationTree tree) {
+        return GET_CONTENT_RESOURCE_METHOD.equals(tree.methodSelect().lastToken().text());
+    }
+
+    private boolean isGetContentResourceUsedOnPage(VariableTree tree) {
+        return tree.initializer() instanceof MethodInvocationTree &&
+            !((MethodInvocationTree) tree.initializer()).symbol().isUnknown() &&
+            isPage(((MethodInvocationTree) tree.initializer()).symbol().owner().type()
+                .fullyQualifiedName()) &&
+            GET_CONTENT_RESOURCE_METHOD.equals(((MethodInvocationTree) tree.initializer()).symbol().name());
+    }
+
+    private boolean isGetContentResourceUsedOnResource(AssignmentExpressionTree tree) {
+        return isResource(tree) &&
+            tree.expression() instanceof MethodInvocationTree &&
+            GET_CONTENT_RESOURCE_METHOD.equals(((MethodInvocationTree) tree.expression()).symbol().name());
+    }
+
+    private boolean isPage(String name) {
+        return Constants.PAGE.equals(name);
+    }
+
+    private boolean isThisAResourceNullCheck(IfStatementTree tree, String ifType) {
+        boolean result = false;
+        if (ifType.equals(tree.condition().kind().name())) {
+            result = isThisANullCheck(tree);
+        }
+        return result;
+    }
+
+    private boolean isRightSideNullCheck(IfStatementTree tree) {
+        return Kind.NULL_LITERAL.equals(tree.condition().lastToken().parent().kind());
+    }
+
+    private boolean isThisANullCheck(IfStatementTree tree) {
+        boolean result = false;
+        if (tree.condition().firstToken().parent() instanceof IdentifierTree &&
+            Constants.RESOURCE_TYPE
+                .equals(((IdentifierTree) tree.condition().firstToken().parent()).symbolType()
+                    .fullyQualifiedName()) &&
+            Kind.NULL_LITERAL.equals(tree.condition().lastToken().parent().kind())) {
+            result = true;
+        } else if (tree.condition().lastToken().parent() instanceof IdentifierTree &&
```

Comment:
SQ issue: "Two branches in a conditional structure should not have exactly the same implementation"

Reply:
Merged from Issue 65 - refactoring proposal #101